internal/infrastructure/tx-builder/covenantless/builder.go

```diff
@@ -1256,11 +1242,25 @@ func (b *txBuilder) extractSweepLeaf(input psbt.PInput) (
 	sweepTapTree := txscript.AssembleTaprootScriptTree(txscript.NewBaseTapLeaf(sweepScript))
 	sweepRoot := sweepTapTree.RootNode.TapHash()
 
+	cosignerFields, err := txutils.GetArkPsbtFields(ptx, inputIndex, txutils.CosignerPublicKeyField)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	if len(cosignerFields) == 0 {
+		return nil, nil, nil, fmt.Errorf("no cosigner pubkeys found")
+	}
+
+	cosignerPubKeys, err := txutils.MakeCosignerPublicKeyList(cosignerFields)
+	if err != nil {
+		return nil, nil, nil, err
+	}
```

Comment:
shall we abstract this code into a single function in `txutils.GetCosignersFromArkPsbt(pts, inputIndex)`?

Reply:
if one doesn't want the func that combines txutils.GetArkPsbtFields + txutils.MakeCosignerPublicKeyList, he would still be free to use these functions... i'm not saying to replace 2 funcs with 1, but to have 1 that calls both